File: web/src/app/shared/models/pocket/creature-templates/creature-template.model.ts
import { Entity } from "../../Entity.model";

export class CreatureTemplateModel extends Entity {
  public maxAttributePoints: number;
  public totalAttributePoints: number;
  public totalSkillsPoints: number;
  public attributes: AttributeTemplateModel[];
  public skills: SkillTemplateModel[];
  public lifes: LifeTemplateModel[];

  constructor() {
    super();
    this.attributes = [];
    this.skills = [];
    this.lifes = [];
  }
}


export class AttributeTemplateModel extends Entity {
}

export class SkillTemplateModel extends Entity {
  public attributeId: string;
  public minorSkills: MinorSkillsTemplateModel[];
  constructor() {
    super();
    this.minorSkills = [];
  }
}

export class MinorSkillsTemplateModel extends Entity {
  public skillTemplateId: string;
}

export class LifeTemplateModel extends Entity {
  public formula: string;
}
export class DefenseTemplateModel extends Entity {
  public formula: string;
}

File: PocketEdition/RoleRollsPocketEdition/CreaturesTemplates/Application/Dtos/CreatureTemplateModel.cs
using RoleRollsPocketEdition.Creatures.Domain;
using RoleRollsPocketEdition.CreaturesTemplates.Domain.Templates;

namespace RoleRollsPocketEdition.CreaturesTemplates.Application.Dtos
{
    public class CreatureTemplateModel
    {

        public CreatureTemplateModel()
        {

        }
        public CreatureTemplateModel(CreatureTemplate template)
        {
            Id = template.Id;
            Name = template.Name;
            TotalAttributePoints = template.TotalAttributePoints;
            MaxAttributePoints = template.MaxAttributePoints;
            TotalSkillsPoints = template.TotalSkillsPoints;
            Attributes = template.Attributes.Select(attribute => new AttributeTemplateModel(attribute)).ToList();
            Skills = template.Skills.Select(skill => new SkillTemplateModel(skill)).ToList();
            Lifes = template.Lifes.Select(life => new LifeTemplateModel(life)).ToList();
        }

        public Guid Id { get; set; }
        public string Name { get; set; }
        public int MaxAttributePoints { get; init; }
        public int TotalAttributePoints { get; set; }
        // 4
        public int TotalSkillsPoints { get; set; }
        public virtual ICollection<AttributeTemplateModel> Attributes { get; set; }
        public ICollection<SkillTemplateModel> Skills { get; set; }

        public ICollection<LifeTemplateModel> Lifes { get; set; }
    }
}
